managedkafka/managedkafka_create_connect_cluster/main.tf

```diff
@@ -46,14 +46,47 @@ resource "google_managed_kafka_connect_cluster" "default" {
   gcp_config {
     access_config {
       network_configs {
-        primary_subnet = "projects/${data.google_project.default.number}/regions/us-central1/subnetworks/default"
+        primary_subnet = google_compute_subnetwork.default.id
       }
     }
   }
 }
 # [END managedkafka_create_connect_cluster]
 
+# [START managedkafka_subnetwork]
+resource "google_compute_subnetwork" "default" {
+  name          = "test-subnetwork"
+  ip_cidr_range = "10.2.0.0/16"
+  region        = "us-central1"
+  network       = google_compute_network.default.id
+
+  provisioner "local-exec" {
+    when        = destroy
+    command     = <<-EOT
+      set -e
+      gcloud compute network-attachments list \
+        --filter="subnetworks:https://www.googleapis.com/compute/v1/${self.id}" \
+        --format="value(name)" --project="${self.project}" |
+        while read -r na_name; do
+          [[ -z "$na_name" ]] && continue
+          for i in {1..5}; do
+            gcloud compute network-attachments delete "$na_name" \
+              --project="${self.project}" --region="${self.region}" --quiet && break
+            if [[ $i -eq 5 ]]; then exit 1; fi
+            sleep 30
+          done
+        done
+    EOT
+    interpreter = ["bash", "-c"]
+  }
+}
```

Comment:
We do not support local-exec in samples. We do not want to encourage our users to use this functionality.

It looks like the rebased merge commit before this had successful tests, so I'm not sure why a new network and subnet are being added, but also why they wouldn't nicely be destroyed. This might be an upstream issue(?). 

If it's a sample that's not nicely testable (the current test takes 80 minutes, where the google_managed_kafka_cluster takes 30 minutes, and the google_managed_kafka_connect_cluster takes 20 minutes to create), we can disable the `terraform apply/destroy` steps.